internal/proxy/privilege_interceptor.go

```diff
@@ -0,0 +1,127 @@
+package proxy
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/milvus-io/milvus/internal/util"
+
+	"github.com/casbin/casbin/v2"
+	"github.com/casbin/casbin/v2/model"
+	jsonadapter "github.com/casbin/json-adapter/v2"
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/util/funcutil"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+type PrivilegeFunc func(ctx context.Context, req interface{}) (context.Context, error)
+
+const (
+	// sub -> role name, like admin, public
+	// obj -> contact object with object name, like Global-*, Collection-col1
+	// act -> privilege, like CreateCollection, DescribeCollection
+	ModelStr = `
+[request_definition]
+r = sub, obj, act
+
+[policy_definition]
+p = sub, obj, act
+
+[policy_effect]
+e = some(where (p.eft == allow))
+
+[matchers]
+m = r.sub == p.sub && globMatch(p.obj, r.obj) && globMatch(p.act, r.act) || r.sub == "admin" || r.act == "All"
+`
+	ModelKey = "casbin"
+)
+
+var modelStore = make(map[string]model.Model, 1)
+
+func initPolicyModel() (model.Model, error) {
+	policyModel, err := model.NewModelFromString(ModelStr)
+	if err != nil {
+		log.Error("NewModelFromString fail", zap.String("model", ModelStr), zap.Error(err))
+		return nil, err
+	}
+	modelStore[ModelKey] = policyModel
+	return modelStore[ModelKey], nil
+}
+
+// UnaryServerInterceptor returns a new unary server interceptors that performs per-request privilege access.
+func UnaryServerInterceptor(privilegeFunc PrivilegeFunc) grpc.UnaryServerInterceptor {
+	initPolicyModel()
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		newCtx, err := privilegeFunc(ctx, req)
+		if err != nil {
+			return nil, err
+		}
+		return handler(newCtx, req)
+	}
+}
+
+func PrivilegeInterceptor(ctx context.Context, req interface{}) (context.Context, error) {
+	if !Params.CommonCfg.AuthorizationEnabled {
+		return ctx, nil
+	}
+	log.Debug("PrivilegeInterceptor", zap.String("type", reflect.TypeOf(req).String()))
+	privilegeExt, err := funcutil.GetPrivilegeExtObj(req)
+	if err != nil {
+		log.Debug("GetPrivilegeExtObj err", zap.Error(err))
+		return ctx, nil
+	}
+	username, err := GetCurUserFromContext(ctx)
+	if err != nil {
+		log.Error("GetCurUserFromContext fail", zap.Error(err))
```

Comment:
The convention is `Error` shall be used when milvus could not handle this situation and needs to quit after this log. While `Warn` means there is something wrong but milvus could still be running after this.

There are lot's of `Error` in this pr, and it will print not needed stack and cause unwanted attention of the users.

Reply:
Looking at some historical code, many parts of the request error are recorded using `log.error`, but after that, milvus is running normally, but the request is returned immediately. For example, if etcd is disconnected, milvus should be running normally at present, but requests related to etcd operations will fail.